Sources/Profile-Actions.php

```diff
@@ -969,4 +969,196 @@ function subscriptions($memID)
 		$context['sub_template'] = 'user_subscription';
 }
 
+/**
+ * Function to allow the user to todo Privacy stuff
+ *
+ * @param int $memID The ID of the member
+ */
+function getProfileData($memID)
+{
+	global $txt, $user_profile, $context, $smcFunc, $user_info;
+
+	if ($memID == $user_info['id'])
+		$context['pdc']['own'] = true;
+	else
+		$context['pdc']['name'] = $context['member']['name'];
+	
+	if (!empty($_REQUEST['nofound']))
+		$context['pdc']['nofound'] = true;
+	
+	if (empty($_REQUEST['activity']))
+		return;
+	
+	$mode = $_REQUEST['activity'];
+
+	$profileData = array();
+	if ($mode == 'profile') // profile
+	{
+		loadMemberData($memID, false, 'profile');
+		$profile = $user_profile[$memID];
+		$removeFields = array('secret_question','tfa_secret','password_salt');
+		foreach ($removeFields as $value)
+		{
+			unset($profile[$value]);
+		}
+		foreach($profile as $key => &$value)
+		{
+			if (is_array($value))
+				$value = $smcFunc['json_encode']($value);
+		}
+		$profileData[0] = array_keys($profile);
```

Comment:
Like you see here: https://github.com/SimpleMachines/SMF2.1/pull/4892#issuecomment-410521610
they get already added.

Reply:
Yes, you are right that there is a header row. Unfortunately, it is filled with numeric keys instead of column names, so it isn't useful at all. Please change that.

Reply:
Nope please read the start post,
you need to apply the pr #5048
otherwise the fetch_all is different across pg and mysql.

Reply:
Fair enough. 🙂